### PR TITLE
Add Validator status

### PR DESCRIPTION
**Pull request by macfarla**
Add method to derive validator status and add this field to ValidatorDuties object
#1348

### CODE REVIEW

File: data/provider/src/main/java/tech/pegasys/artemis/api/ChainDataProvider.java [hunk 1]
```diff
@@ -59,6 +61,29 @@ public ChainDataProvider(
...
+      return ValidatorStatus.PENDING;
+    }
+    if (validator.getExit_epoch().equals(Constants.FAR_FUTURE_EPOCH)) {
+      return ValidatorStatus.ACTIVE;
+    }
+    if (epoch.compareTo(validator.getExit_epoch()) < 0) {
+      if (validator.isSlashed()) {
+        return ValidatorStatus.SLASHING;
+      } else {
+        return ValidatorStatus.EXITING;
```
Comment (rolfyone) [inline data/provider/src/main/java/tech/pegasys/artemis/api/ChainDataProvider.java]: i'd be super tempted to use a ternary here
```
return validator.isSlashed() ? SLASHING : EXITING;
```

File: data/provider/src/main/java/tech/pegasys/artemis/api/ChainDataProvider.java [hunk 2]
```diff
@@ -59,6 +61,29 @@ public ChainDataProvider(
...
 
+  public static ValidatorStatus getValidatorStatus(Validator validator, UnsignedLong epoch) {
+    if (epoch.compareTo(validator.getActivation_eligibility_epoch()) < 0) {
+      return ValidatorStatus.DEPOSITED;
+    }
+    if (epoch.compareTo(validator.getActivation_epoch()) < 0) {
+      return ValidatorStatus.PENDING;
+    }
+    if (validator.getExit_epoch().equals(Constants.FAR_FUTURE_EPOCH)) {
+      return ValidatorStatus.ACTIVE;
```
Comment (rolfyone) [inline data/provider/src/main/java/tech/pegasys/artemis/api/ChainDataProvider.java]: nit: can probably just static import?

File: data/provider/src/main/java/tech/pegasys/artemis/api/ChainDataProvider.java [hunk 3]
```diff
@@ -59,6 +61,29 @@ public ChainDataProvider(
     this.chainStorageClient = chainStorageClient;
   }
 
+  public static ValidatorStatus getValidatorStatus(Validator validator, UnsignedLong epoch) {
```
Comment (rolfyone) [inline data/provider/src/main/java/tech/pegasys/artemis/api/ChainDataProvider.java:71]: probably want to check validator and return unknown if null before we start this whole process..

Comment (macfarla) [reply data/provider/src/main/java/tech/pegasys/artemis/api/ChainDataProvider.java:71]: done

File: data/provider/src/main/java/tech/pegasys/artemis/api/ChainDataProvider.java (hunk 2 above)
Comment (macfarla) [reply data/provider/src/main/java/tech/pegasys/artemis/api/ChainDataProvider.java]: done

File: data/provider/src/main/java/tech/pegasys/artemis/api/ChainDataProvider.java (hunk 1 above)
Comment (macfarla) [reply data/provider/src/main/java/tech/pegasys/artemis/api/ChainDataProvider.java]: done

**Review by rolfyone — APPROVED** [review]
LGTM

**Pull request merged**